prow/cmd/deck/main.go

```diff
@@ -462,6 +464,31 @@ func handleBadge(ja *jobs.JobAgent) http.HandlerFunc {
 	}
 }
 
+func handleJobHistory(templateRoot string, ca *config.Agent, gcsClient *storage.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		tmpl, err := getJobHistory(r.URL, ca.Config(), gcsClient)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get job history: %v", err)
+			logrus.Error(msg)
+			http.Error(w, msg, http.StatusInternalServerError)
+			return
+		}
+		t := template.New("job-history.html")
+		if _, err := prepareBaseTemplate(templateRoot, ca, t); err != nil {
+			fmt.Fprintf(w, "error preparing base template: %v", err)
```

Comment:
`http.Error()` followed by `return` is probably more appropriate for this error and the following ones.